tests/reduction/reduction_without_identity_param_common.h

```diff
@@ -0,0 +1,361 @@
+    cgh.parallel_for<kernel<VariableT, 2>>(range, reduction, lambda);
+  });
+  if (output_buffer.get_host_access()[0] != expected_value) {
+    log.fail(get_fail_message(type_name, output_buffer.get_host_access()[0],
+                              expected_value),
+             __LINE__);
+  }
+}
+
+/** @brief Run test for reduction that receive sycl::span for construct
+ *         reducer object
+ *  @tparam VariableT Variable type from type coverage
+ *  @tparam UseCombineFlagT std::integral_constant type that let switch between
+ *          calling .combine() function or operator +, *, ^= e.t.c.
+ *  @tparam UsePropertyFlagT UseCombineFlagT std::integral_constant type that
+ *          let switch between using and don't using
+ *          sycl::property::reduction::initialize_to_identity
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam RangeT sycl::range or sycl::nd_range type
+ *  @param functor The functor (plus, multiplies e.t.c) with which the test runs
+ *  @param range sycl::range or sycl::nd_range type object
+ *  @param queue sycl::queue class object
+ *  @param log sycl_cts::util::logger class object
+ *  @param typeName a string representing the currently tested type
+ */
+template <typename VariableT, typename UseCombineFlagT,
+          typename UsePropertyFlagT, typename FunctorT, typename RangeT>
+void run_test_for_span(FunctorT functor, RangeT range, sycl::queue &queue,
+                       sycl_cts::util::logger &log,
+                       const std::string &type_name) {
+  if (!reduction_common::check_usm_shared_aspect(queue, log)) {
+    return;
+  }
+  sycl::buffer<VariableT> initial_buf{
+      reduction_common::get_buffer<VariableT>()};
+  VariableT expected_value{reduction_common::get_expected_value(
+      functor, initial_buf,
+      reduction_common::get_init_value_for_expected_value<VariableT, FunctorT>())};
+  auto allocated_memory{
+      usm_helper::allocate_usm_memory<sycl::usm::alloc::shared, VariableT>(
+          queue, number_elements)};
+  auto value_for_filling{
+      reduction_common::get_init_value_for_reduction<VariableT, FunctorT, UsePropertyFlagT>()};
+  for (int i{}; i < number_elements; i++) {
+    allocated_memory.get()[i] = value_for_filling;
+  }
+
+  queue.submit([&](sycl::handler &cgh) {
+    sycl::span<VariableT, number_elements> span(allocated_memory.get(),
+                                                number_elements);
+    auto reduction{get_reduction_for_span<UsePropertyFlagT>(span, functor)};
+    auto lambda{
+        reduction_get_lambda::get_lambda_for_span<VariableT, RangeT,
+                                                  UseCombineFlagT, FunctorT>(
+            initial_buf.template get_access<sycl::access_mode::read>(cgh),
+            number_elements)};
+    cgh.parallel_for<kernel<VariableT, 3>>(range, reduction, lambda);
+  });
+  for (int i{}; i < number_elements; i++) {
```

Comment:
This code will give a warning like this:

comparison of integers of different signs: 'int' and 'const size_t' (aka 'const unsigned long') [-Wsign-compare]

Reply:
I've replace using `int` to `size_t`, this should remove this warning
I've made this change in this commit: b1f4fcfe1751d94797b47a43abaa20d8f5cf8106